technology-considerations/s3-user-access-to-objects.md

```diff
@@ -0,0 +1,121 @@
+# Restricting User Access To S3 Objects
+
+## Problem
+
+Ensure that TDR users are not able to overwrite/view another user’s objects in the TDR AWS S3 upload bucket.
+
+Potential options to resolve the problem are to use:
+* AWS Cognito;
+* Signed URLS.
+
+The purpose of the spike was to see whether using AWS Cognito was a feasible option.
+
+AWS Cognito has some advantages over using signed URLs:
+* Using signed URLs means an eavesdropper, potentially, can grab the signed url and use it until it expires
+* AWS Cognito will work with AWS KMS encrypted AWS S3, can set up an AWS IAM or key policy allowing access to the AWS KMS key from AWS Cognito identities
+
+### Additional Considerations
+
+Unclear how AWS Cognito will work with Javascript silent login using Keycloak's Javascript library as the tokens are stored in Redis and not cookies.
```

Comment:
Keycloak's silent login lets us log the users in on the frontend and backend simultaneously, with different Keycloak clients. Sam got this working in the prototype, and he's added it to the new frontend: https://github.com/nationalarchives/tdr-transfer-frontend/pull/57

Reply:
Removed this